products/views.py

```diff
@@ -1,3 +1,42 @@
-from django.shortcuts import render
+from django.http.response        import Http404, JsonResponse
+from django.views                import View
+from django.db.models            import Value
+from django.db.models.functions  import Coalesce
+from django.db.models.aggregates import Count, Avg
+from django.shortcuts            import get_object_or_404
 
-# Create your views here.
+from products.models             import Product
+from users.utils                 import login_decorator
+
+class ProductDetailView(View):
+    @login_decorator
+    def get(self, request, product_id):
+        try:
+            product = get_object_or_404(Product, id=product_id)
+            user    = request.user
+
+            sart_point_info = product.review_set.aggregate(
+                    avg=Coalesce(Avg('star_point'), Value(0.0)),
+                    count=Count('id'))
```

Comment:
```suggestion
            sart_point = product.review_set.aggregate(
                    avg=Coalesce(Avg('star_point'), Value(0.0)),
                    count=Count('id'))
```
- star_point로도 충분히 어떤 의미인지 알 수 있을 것 같습니다.
- avg, count의 정보를 담고 있는 의미로 사용한거면 information으로 줄이지 말고 사용해 주세요.
- Coalesce의 왜 사용 했고, 어떻게 사용하는지 코멘트로 남겨주세요.

Reply:
> Coalesce의 왜 사용 했고, 

리뷰에 대한 별점을 나타내기 위해서  해당 상품에 리뷰에 대한  별점의 평균을 내야되었습니다.
일반적으로는 Coalesce가 없어도 잘 됩니다. 다만 상품에 리뷰가 하나라도 없는경우 `null`를 리턴합니다.
저는 상품에 대한 리뷰가 1개도 없을시에도 0으로 나와야 된다고 생각하여서 사용하였습니다.

> 어떻게 사용하는지 코멘트로 남겨주세요

Coalesce(a, b, c, d, 0)와 같이 2개 이상의 인자를 받고, 왼쪽부터 차례대로 검사하여 `null`이 아닌 첫번째 값을 리턴합니다.
따라서 위에같은 경우 평점이 `null`이 될 시에 그 다음 값인 0.0을 반환할 수 있도록 사용 할 수 있었습니다.

Reply:
star_point 로 변수명 변경 완료했습니다.

Reply:
삼항연산자로 None일 경우 0을 반환하는 식의 코드가 Coalesce를 사용했을 때 보다, 코스트 측면에서 더 싸다고 구두로 말씀해주셔서 수정 완료했습니다.